x-pack/plugins/secops/public/components/range_date_picker/quick_select_popover/quick_select.tsx

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiButton,
+  EuiFieldNumber,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiSelect,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import { getOr } from 'lodash/fp';
+import moment, { Moment } from 'moment';
+import React from 'react';
+import { pure } from 'recompose';
+import { DateType, RecentlyUsedI } from '..';
+
+interface Options {
+  value: string;
```

Comment:
Optionally you can constrain your types if you want to be specific like an enum you did in other areas such as DatePickerOptions to constrain the strings to specific names.